bindings/wallet-core/src/wallet.rs

```diff
@@ -135,24 +144,61 @@ impl Wallet {
         self.icarus.dump_in(&mut dump);
 
         let (ignored, transactions) = dump.finalize();
+        let mut raws = Vec::with_capacity(transactions.len());
 
-        let transactions = transactions
-            .into_iter()
-            .map(|t| t.serialize_as_vec().unwrap())
-            .collect();
+        for (used, f) in transactions {
+            let id = f.id();
+
+            self.pending_transactions.insert(id, used);
+            raws.push(f.serialize_as_vec().unwrap());
+        }
 
         Conversion {
             ignored,
-            transactions,
+            transactions: raws,
+        }
+    }
+
+    /// use this function to confirm a transaction has been properly received
+    ///
+    /// This function will automatically update the state of the wallet
+    pub fn confirm_transaction(&mut self, id: FragmentId) {
+        if let Some(inputs) = self.pending_transactions.remove(&id) {
+            for input in inputs {
+                match input.to_enum() {
+                    InputEnum::UtxoInput(pointer) => {
+                        self.icarus.remove(pointer);
+                        self.daedalus.remove(pointer);
+                    }
+                    InputEnum::AccountInput(identifier, value) => {
+                        self.account.remove(identifier, value);
+                    }
+                }
+            }
         }
     }
 
+    /// get access to all the pending transaction
+    ///
+    /// TODO: this might need to be updated to have a more user friendly
+    ///       API. Currently do this for simplicity
+    pub fn pending_transactions(&self) -> &HashMap<FragmentId, Vec<Input>> {
+        &self.pending_transactions
+    }
+
+    /// remove a given pending transaction returning the associated Inputs
+    /// that were used for this transaction
+    pub fn remove_pending_transaction(&mut self, id: &FragmentId) -> Option<Vec<Input>> {
```

Comment:
why do we expose this? I'm not sure how it will be used. Will be add later a new function to try to resend those inputs?

Reply:
It's more about awaiting a notification for a transaction that has been successfully added to the blockchain.